prow/plugins/trigger/ic.go

```diff
@@ -155,22 +155,18 @@ func handleIC(c client, trustedOrg string, ic github.IssueCommentEvent) error {
 		return err
 	}
 
+	// Determine if any branch-shard of a given job runs against the base branch.
+	anyShardRunsAgainstBranch := map[string]bool{}
+	for _, job := range requestedJobs {
+		if job.RunsAgainstBranch(pr.Base.Ref) {
+			anyShardRunsAgainstBranch[job.Context] = true
+		}
+	}
+
 	var errors []error
 	for _, job := range requestedJobs {
-		build := true
 		if !job.RunsAgainstBranch(pr.Base.Ref) {
-			build = false
-		} else if !job.AlwaysRun && job.RunIfChanged != "" {
-			changes, err := files()
-			if err != nil {
-				return err
-			}
-			if !job.RunsAgainstChanges(changes) {
-				build = false
-			}
-		}
-		if !build {
-			if !job.SkipReport {
+			if !job.SkipReport && !anyShardRunsAgainstBranch[job.Context] {
```

Comment:
Isn't `!anyShardRunsAgainstBranch[job.Context]` always going to be `true` at this point?

Reply:
Not necessarily, we don't check that the jobs run against the current branch before adding them to the `requestedJobs` list because we may still need to mark them as 'Skipped'. So it is possible that no requested versions of a given job run on the branch.

Reply:
Adding to what @cjwagner said: we definitely have jobs that need a skipped status marked that are also sharded, the gpu job only runs against 1.7+ and has different 1.7,1.8, 1.9/master versions last I checked.